sender/src/lib.rs

```diff
@@ -0,0 +1,131 @@
+#![no_std]
+
+use codec::Output;
+use core::fmt::Debug;
+use core::marker::PhantomData;
+use gstd::{errors::Error as GStdError, msg::MessageFuture, prelude::*, ActorId, MessageId};
+use parity_scale_codec::{Decode, Error as ParseError};
+
+#[derive(Default)]
+struct SendArgs {
+    value: u128,
+    reply_deposit: u64,
+}
+
+#[derive(Default)]
+pub struct GStdSender;
+
+impl GStdSender {
+    pub fn new() -> Self {
+        Self
+    }
+}
+
+impl GStdSender {
+    pub fn send(
+        &self,
+        address: ActorId,
+        payload: Vec<u8>,
+        value: u128,
+        reply_deposit: u64,
+    ) -> Result<MessageFuture, GStdError> {
+        let future = gstd::msg::send_bytes_for_reply(address, payload, value, reply_deposit)?;
+
+        Ok(future)
+    }
+}
+
+pub struct Call<'a, R: Decode + Debug> {
+    /// serialized method and args
+    payload: Vec<u8>,
+    /// optional args
+    send_args: SendArgs,
+    /// the client to send the message
+    sender: &'a GStdSender,
+    /// silence the compiler
+    _marker: PhantomData<R>,
+}
+
+#[non_exhaustive]
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum SendError {
+    Parse(ParseError),
+    Sender(GStdError),
+}
+
+impl From<ParseError> for SendError {
+    fn from(e: ParseError) -> Self {
+        Self::Parse(e)
+    }
+}
+
+impl From<GStdError> for SendError {
+    fn from(e: GStdError) -> Self {
+        Self::Sender(e)
+    }
+}
+
+impl<'a, R: Decode + Debug> Call<'a, R> {
+    pub fn new<T: Encode + Debug>(method: &str, args: T, client: &'a GStdSender) -> Self {
+        let capacity = method.len() + 1 + args.encoded_size();
+        let mut payload = Vec::with_capacity(capacity);
+        payload.extend_from_slice(method.as_bytes());
+        payload.push_byte(b'/');
```

Comment:
Encoding is wrong. It has to be, there are no any slashes
`|encoded-method-name||encoded-payload|`
See example here https://github.com/gear-tech/sails/blob/cc3a4a4f29eb7123216ab3ff6361a96a032482a1/examples/rmrk/resource/wasm/tests/resources.rs#L203